specifications/bcastByz/bcastByz.tla

```diff
@@ -0,0 +1,983 @@
+------------------------------ MODULE bcastByz ------------------------------
+
+(*
+ * TLA+ encoding of a parameterized model of the broadcast distributed  
+ * algorithm with Byzantine faults.
+ *
+ * This is a one-round version of asynchronous reliable broadcast (Fig. 7) from:
+ *
+ * T. K. Srikanth, Sam Toueg. Simulating authenticated broadcasts to derive
+ * simple fault-tolerant algorithms. Distributed Computing 1987,
+ * Volume 2, Issue 2, pp 80-94
+ *                                                           
+ * A short description of the parameterized model is described in: Gmeiner,   
+ * Annu, et al. "Tutorial on parameterized model checking of fault-tolerant   
+ * distributed algorithms." International School on Formal Methods for the  
+ * Design of Computer, Communication and Software Systems. Springer  
+ * International Publishing, 2014.                   
+ *
+ * This specification has a TLAPS proof for property Unforgeability: if process p 
+ * is correct and does not broadcast (p, m, k), then no correct process ever 
+ * accepts (p, m, k). The formula InitNoBcast represents that the transmitter 
+ * does not broadcast any message. So, our goal is to prove the  formula
+ *      (InitNoBcast /\ [][Next]_vars) => Unforg                    
+ *
+ * Igor Konnov, Thanh Hai Tran, Josef Widder, 2016
+ *
+ * This file is a subject to the license that is bundled
+ * together with this package and can be found in the file LICENSE.
+ *)
+
+
+EXTENDS Naturals, FiniteSets
+
+CONSTANTS N, T, F
+
+VARIABLE Corr           (* The correct processes *)
+VARIABLE Faulty         (* The faulty processes *)
+                        (* Corr and Faulty are declared as variables since we want to 
+                           check all possible cases. And after the initial step, Corr
+                           and Faulty are unchanged. *)
+VARIABLE pc             (* the control state of each process *)
+VARIABLE rcvd           (* the messages received by each process *)
+VARIABLE sent           (* the messages sent by all correct processes *)
+VARIABLE newMessages    (* the message received at the current step by each process *)
+
+ASSUME NTF == N \in Nat /\ T \in Nat /\ F \in Nat /\ (N > 3 * T) /\ (T >= F) /\ (F >= 0)
+
+Proc == 1 .. N                 (* all processess, including the faulty ones    *)
+ByzMsgs == { <<p, "ECHO">> : p \in Faulty }                            
+
+
+M == { "ECHO" }
+
+vars == << pc, rcvd, sent, newMessages, Corr, Faulty >>
+
+(* Both correct and faulty processes send only <ECHO> msgs. *)
+Receive(self) ==
+  /\ newMessages' \in SUBSET ( sent \cup ByzMsgs )
+  /\ rcvd' = [ i \in Proc |-> IF i # self THEN rcvd[i] ELSE rcvd[self] \cup newMessages' ]
+                             
+(* If received an INIT message and not send <ECHO> before,    
+ * then send <ECHO> to all - line 3.                                        
+ * UponV1 is for the 1st ITE.
```

Comment:
"UponV1 is for the 1st ITE" is unclear. Perhaps you want to show the pseudo-code that you refer to?